hack/demo/generate-asciidemo.sh

```diff
@@ -13,45 +13,79 @@ function check_prereq() {
     fi
 }
 
-function cleanup() {
-    if [ -d $WKDIR ]
-    then
+cleanup() {
+    if [[ -n "${WKDIR-}" && -d $WKDIR ]]; then
         rm -rf $WKDIR
     fi
 }
 
-function usage() {
-    echo "$0 [options]"
-    echo "where options is"
-    echo " h  help (this message)"
+usage() {
+    echo "$0 [options] <demo_script>"
+    echo ""
+    echo "example:"
+    echo "  # Generate asciinema demo described by gzip-demo-script.sh to gzip-demo-script.cast"
+    echo "  $0 gzip-demo-script.sh"
+    echo ""
+    echo "  # Generate asciinema demo described by demo-script.sh to catalogd-demo.cast"
+    echo "  $0 -n catalogd-demo demo-script.sh"
+    echo ""
+    echo "  # Generate and upload catalogd-demo.cast"
+    echo "  $0 -u -n catalogd-demo demo-script.sh"
+    echo ""
+    echo "options:"
+    echo "  n  demo name"
+    echo "  u  upload cast (default: false)"
+    echo "  h  help (this message)"
     exit 1
 }
 
 set +u
-while getopts 'h' flag; do
+while getopts ':hn:u' flag; do
     case "${flag}" in
-        h) usage ;;
+        h)
+            usage
+            ;;
+        n)
+            DEMO_NAME="${OPTARG}"
+            ;;
+        u)
+            UPLOAD=true
+            ;;
+        :)
+            echo "Error: Option -${OPTARG} requires an argument."
+            usage
+            ;;
+        \?)
+            echo "Error: Invalid option -${OPTARG}"
+            usage
+            ;;
     esac
-    shift
 done
+shift $((OPTIND - 1))
 set -u
 
+DEMO_SCRIPT="${1-}"
+
+if [ -z $DEMO_SCRIPT ]; then
+  usage
+fi
+
 WKDIR=$(mktemp -d -t generate-asciidemo.XXXXX)
-if [ ! -d ${WKDIR} ]
-then
+if [ ! -d ${WKDIR} ]; then
```

Comment:
😁 someone doesn't like dropped-then conditionals ... 🎶  😄

Reply:
just preference on my side - happy to revert if it's better form ^^